### ISSUE TITLE
Restore keys from the key backup for the displayed room

### ISSUE THREAD

**Issue by manuroe** [vector-im/element-x-ios#417]
Without restoring the keys to decrypt messages in the rooms, the app looks quite empty just after logging in.
Now, that we have device verification working, we could try to automatically restore some keys from the key backup.

Current Element apps restore all the keys from the key backup in a single request. We probably have users reaching 1M of keys. This operation is slow and expensive in bandwidth, import processing and device storage.

With EX, we could approach the keys restoration in a more optimised way. As a start, we could only restore keys of a room only when the user opens it.

Some API needs to be exposed through the SDK:
- Request 4S keys via gossiping
- Restore keys for a given room

Ideally the app should restore keys only once and should not make the request on each room opening. That part is probably more an application level logic.

**Comment by stefanceriu** [comment]
> backups are only implemented in the crypto crate, no API exists for the top level crate

**Comment by stefanceriu** [comment]
Superseded by https://github.com/vector-im/element-internal/issues/516
